### PR TITLE
[SKOOP-131] Components to display, create, update publications

### PR DESCRIPTION
**Pull request by svetivanova**
Components to display, create, update publications
tests

@LebedkoDmitry could you please take a look?

### CODE REVIEW

File: src/app/publications/publications-edit.component.html [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    <div fxFlex>
+      <mat-form-field class="publications-edit-form-field">
+        <input matInput [matDatepicker]="publicationDatePicker" [max]="maxDate"  placeholder="Date" formControlName="date">
+        <mat-datepicker-toggle matSuffix [for]="publicationDatePicker"></mat-datepicker-toggle>
+        <mat-datepicker #publicationDatePicker></mat-datepicker>
+      </mat-form-field>
+    </div>
+    <div fxFlex>
+      <mat-form-field class="publications-edit-form-field">
+        <input matInput  placeholder="Publisher or location of confluence, etc." formControlName="publisher" [required]="true">
```
Comment (LebedkoDmitry) [inline src/app/publications/publications-edit.component.html]: There should be `conference` instead of `confluence`.

File: src/app/publications/publications-new.component.html [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    <div fxFlex>
+      <mat-form-field class="publications-new-form-field">
+        <input matInput [matDatepicker]="publicationDatePicker" [max]="maxDate"  placeholder="Date" formControlName="date">
+        <mat-datepicker-toggle matSuffix [for]="publicationDatePicker"></mat-datepicker-toggle>
+        <mat-datepicker #publicationDatePicker></mat-datepicker>
+      </mat-form-field>
+    </div>
+    <div fxFlex>
+      <mat-form-field class="publications-new-form-field">
+        <input matInput  placeholder="Publisher or location of confluence, etc." formControlName="publisher" [required]="true">
```
Comment (LebedkoDmitry) [inline src/app/publications/publications-new.component.html]: There should be `conference` instead of `confluence`.

File: src/app/publications/publications.component.html [hunk 3]
```diff
@@ -0,0 +1,50 @@
+<div class="mat-typography">
+  <div *ngIf="errorMessage" class="publications-error-message">{{errorMessage}}</div>
+  <div class="publications-toolbar">
+    <button mat-raised-button (click)="openNewDialog()" aria-label="Add testimonial">Add publication</button>
```
Comment (LebedkoDmitry) [inline src/app/publications/publications.component.html]: Please change `aria-label` from `Add testimonial` to `Add publication`.

File: src/app/publications/publications.component.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+      .afterDismissed().subscribe((publication: PublicationResponse) => {
+      if (publication) {
+        this.loadPublications();
+      }
+    });
+  }
+
+  openEditDialog(publication: PublicationResponse) {
+    this.bottomSheet.open(PublicationsEditComponent, {
+      data: <PublicationResponse>{
```
Comment (LebedkoDmitry) [inline src/app/publications/publications.component.ts]: I suggest to use `Object.assign({}, publication)` here. What do you think?

File: src/app/util/util.spec.ts [hunk 5]
```diff
@@ -122,4 +123,26 @@ describe('Util', () => {
       });
   });
 
+  describe('.dateIsInPast()', () => {
```
Comment (LebedkoDmitry) [inline src/app/util/util.spec.ts]: I suggest to add test to verify that date is in past is not allowed (a negative case). What do think?

File: src/app/publications/publications-edit.component.html (hunk 1 above)
Comment (svetivanova) [reply src/app/publications/publications-edit.component.html]: thank you. done

File: src/app/publications/publications-new.component.html (hunk 2 above)
Comment (svetivanova) [reply src/app/publications/publications-new.component.html]: thank you. done

File: src/app/publications/publications.component.html (hunk 3 above)
Comment (svetivanova) [reply src/app/publications/publications.component.html]: thank you. done

File: src/app/publications/publications.component.ts (hunk 4 above)
Comment (svetivanova) [reply src/app/publications/publications.component.ts]: thank you. done

File: src/app/util/util.spec.ts (hunk 5 above)
Comment (svetivanova) [reply src/app/util/util.spec.ts]: thank you. done
